### PR TITLE
Fix active validator DB api; Correct validator snapshots; Finish commission rate check

### PR DESCRIPTION
**Pull request by rlan35**
Also added basic staking wireframes for block body.

and more sanity check on commission rate of validators

and rpcs for validator addresses and validator info.

### CODE REVIEW

File: core/types/bodyv2.go [hunk 1]
```diff
@@ -51,6 +51,15 @@ func (b *BodyV2) TransactionAt(index int) *Transaction {
 	return b.f.Transactions[index].Copy()
 }
 
+// StakingTransactionAt returns the staking transaction at the given index in this block.
+// It returns nil if index is out of bounds.
+func (b *BodyV2) StakingTransactionAt(index int) *staking.StakingTransaction {
+	if index < 0 || index >= len(b.f.StakingTransactions) {
```
Comment (fxfactorial) [inline core/types/bodyv2.go]: `.f` is never nil?

File: staking/types/delegation.go [hunk 2]
```diff
@@ -93,3 +93,20 @@ func (d *Delegation) DeleteEntry(epoch *big.Int) {
 		d.Entries = entries
 	}
 }
+
+// RemoveUnlockedUndelegations removes all fully unlocked undelegations and returns the total sum
+func (d *Delegation) RemoveUnlockedUndelegations(curEpoch *big.Int) *big.Int {
+	totalWithdraw := big.NewInt(0)
+	count := 0
+	for j := range d.Entries {
+		if curEpoch.Cmp(d.Entries[j].Epoch) > 14 { // need to wait at least 14 epochs to withdraw;
```
Comment (fxfactorial) [inline staking/types/delegation.go]: maybe the 14 should be a named const? like in staking/effective

Comment (fxfactorial) [reply staking/types/delegation.go]: oh wait, the Cmp gives back 1, 0, -1 right, so not the > 14

Comment (rlan35) [reply staking/types/delegation.go]: Goot catch. Done.

File: core/types/bodyv2.go (hunk 1 above)
Comment (rlan35) [reply core/types/bodyv2.go]: It's a struct, so it can't be nil

**Pull request merged**